### ISSUE TITLE
Map Themes - default behaviour for excluded layers

### ISSUE THREAD

**Issue by aharfoot** [qgis/QGIS#50106]
### Feature description

If a layer is added to an project in QGIS with map themes defined it will not be referenced by those themes. When an existing map theme is applied in the map or layout window, then unreferenced layers are hidden. This is currently the default behaviour in 3.22.10

I would like to suggest that the default behaviour for unreferenced layers be changed so that their visibility setting is **not** altered when the map theme is applied. This would permit the map theme to control a set of layers in the project, whilst leaving other layers to be configured independently. A good use case would be the control of the primary map content through a map theme, but the choice of basemapping being made independently.

**Comment by roya0045** [comment]
And then you'd have to change all your themes again to disable new layers? I don't think that's a good idea.

**Comment by aharfoot** [comment]
No, I'm not saying that unreferenced layers should be enabled when a map theme is applied, I'm saying that they shouldn't have their visibility state altered. So if you didn't want new layers to be visible, you would toggle their visibility off with no changes to existing themes required.

**Comment by roya0045** [comment]
Exactly, so you'd have two systems without any way to know what is considered or not. Thus you're forced to redo your themes to have them all accounted for in some cases.

I get your point, but the goal of a theme is to show the map as when the theme was made/updated. Having random layers not handled just make them less reliable and complicate the logistics.

**Comment by aharfoot** [comment]
There are situations where this change might increase the overhead in maintaining map themes, and those where it would dramatically reduce it.

Map themes fall into the 'invest time up front to save in subsequent reuse category' which to me supports increasing their versatility over potentially increasing the time taken to create or manage them.

There are clear use cases for partial map themes, and changing the default behaviour of unreferenced map layers would be necessary to implement them. The natural complement to this change would be to add a mechanism to display the subset of layers participating in any given theme (possibly by selecting them in the layer list when the theme is activated), and to be able to create or update a partial map theme (by having a layer list selection active when replacing the theme?)
